va/va.h

```diff
@@ -1137,6 +1142,17 @@ typedef union _VAConfigAttribValMultipleFrame {
     uint32_t value;
 }VAConfigAttribValMultipleFrame;
 
+/** brief Attribute value VAConfigAttribValContestPriority */
+typedef union _VAConfigAttribValContextPriority{
+    struct{
+        /** \brief the priority , for the Query operation it will return the range of priority */
+        uint32_t priority     :16;
```

Comment:
I assume this "for the Query operation it will return the range of priority" sounds weird. We don't have ranges here, we have only one value. If it's 'max' I propose to say it explicitly in the code comment that range is between [0, priority].

Reply:
echo @dmitryermilov. It is not clear what range of priority means and how it can be queried? We have couple of choices here:
1. Declare in VAAPI spec that we support range of priorities as [0, max] where 0 is lowest priority, max is a highest and this highest value can be returned by the query. Downside is that this will be VAAPI priority declaration which driver will need to map to OS specifics.
2. Same as 1), but declare [min, max]. I this case we need to expose 2 attributes, one for min another for max
3. Again, have [min, max], but have structure for that (or different bits for min and for max)

I would go with 1).

Reply:
In the graphics driver (mesa), the priority is defined three priority (LOW(-512), DEFAULT(0), HIGH(512)). We should consider the alignment between media and graphic, even computing for some user scenario.

Reply:
And KMD has -1023 to 1023 as I heard. I wonder how mesa maps to KMD in this case? is the strict = and they effectively miss part of the range? or the remap 512 to 1023?

We are not mesa, we have own interface and should consider what aligns with our own API style and what makes sense to us. Whenever possible we eventually should align with others, but max priority makes more sense to me in case of libva.

Reply:
In the mesa, it just map LOW/MEDIUM/HIGH to three KMD priority values (-512/0/512). You could say it dropped the range, just picked three typical values.
#define GEN_CONTEXT_LOW_PRIORITY ((I915_CONTEXT_MIN_USER_PRIORITY-1)/2)
#define GEN_CONTEXT_MEDIUM_PRIORITY (I915_CONTEXT_DEFAULT_PRIORITY)
#define GEN_CONTEXT_HIGH_PRIORITY ((I915_CONTEXT_MAX_USER_PRIORITY+1)/2)

Yes, we are own the interface. In the MediaSDK, we also use three priority types other than a range for the user. Itself is already aligned with Mesa definition. 
Back to VA interface, it become a simple question.
1) expose three typical priority values to user, like mesa.
     Less capability for user, more straightforward, low risk when resource competing across component. 
2) expose a range to user, user can define their priority strategy. 
     More capability for user, user should carefully define the priority value.

Reply:
Thank you for insight, @hongbolv. One more question: do I understand correctly that -1023, 1023 range is i915 defined one. So, some other KMD may potentially have some other range?

Reply:
Personally prefer to expose a range [min, max] to the VAAPI user.

Reply:
Yes, the  range (-1023, 1023) is i915 defined.

Reply:
This addressed my previous concern:
•	for the Query operation (read) it represent highest priority
•	for the set operation (write), value should be [0~highest priority] , 0 is lowest priority.

Thank you.